core/commons/che-core-commons-gwt/src/main/java/org/eclipse/che/api/promises/client/js/JsPromiseProvider.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.promises.client.js;
+
+import elemental.js.util.JsArrayOf;
+import elemental.util.ArrayOf;
+
+import com.google.gwt.core.client.JavaScriptObject;
+import com.google.gwt.core.client.JsArrayMixed;
+
+import org.eclipse.che.api.promises.client.Promise;
+import org.eclipse.che.api.promises.client.PromiseError;
+import org.eclipse.che.api.promises.client.PromiseProvider;
+
+/**
+ * @author Vlad Zhukovskyi
+ */
+public class JsPromiseProvider implements PromiseProvider {
+    @Override
+    public native <V> Promise<V> create(Executor<V> conclusion) /*-{
```

Comment:
`conclusion` - strange name, as for me.
I'd use `executor` instead of `conclusion` there.